modules/collect/src/main/java/com/opengamma/strata/collect/Guavate.java

```diff
@@ -303,6 +302,58 @@ public R next() {
     };
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Function used in a stream to cast instances to a particular type without filtering.
+   * <p>
+   * This method returns a function that can be used with {@link Stream#map(Function)}
+   * to cast elements in a stream to a particular type, throwing an exception if any
+   * element is not of the specified type.
+   * <p>
+   * This would be used as follows (with a static import):
+   * <pre>
+   *   stream.map(casting(Foo.class));
+   * </pre>
+   * <p>
+   * This replaces code of the form:
+   * <pre>
+   *   stream.map(Foo.class::cast);
+   * </pre>
+   *
+   * @param <T>  the type of element in the input stream
+   * @param <R>  the type of element in the output stream
+   * @param cls  the type of element in the output stream
+   * @return the function
+   */
+  public static <T, R extends T> Function<T, R> casting(Class<R> cls) {
```

Comment:
What's the benefit of using this over the `Foo.class::cast` style?

Reply:
Teaching people to use methods in Guavate seems to be more successful than teaching them this particular method reference approach. (Many don't appreciate that the LHS of a method ref can be an expression)